types/operations/comparisons/ComparisonID.hpp

```diff
@@ -54,6 +54,28 @@ extern const char *kComparisonShortNames[
     static_cast<typename std::underlying_type<ComparisonID>::type>(
         ComparisonID::kNumComparisonIDs)];
 
+/**
+ * Flips a comparison. As in greater than flips to less than, less than flips
+ * to greater than, and similarly for greater/less than or equals. Notice that
+ * flipping equals results in equals, same for not equals.
+ * @param comparison - The Id of a comparison to flip.
+ * @return The flipped comparison id.
+ */
+inline ComparisonID flipComparisonID(const ComparisonID comparison) {
```

Comment:
That's a function, Zuyu, not an instance method. It can't be marked const.

On Wed, Jan 27, 2016 at 10:21 AM Zuyu ZHANG <notifications@github.com>
wrote:

> In types/operations/comparisons/ComparisonID.hpp
> <https://github.com/pivotalsoftware/quickstep/pull/18#discussion_r51025581>
> :
>
> > @@ -54,6 +54,28 @@ extern const char *kComparisonShortNames[
> >      static_cast<typename std::underlying_type<ComparisonID>::type>(
> >          ComparisonID::kNumComparisonIDs)];
> >
> > +/**
> > + * Flips a comparison. As in greater than flips to less than, less than flips
> > + * to greater than, and similarly for greater/less than or equals. Notice that
> > + * flipping equals results in equals, same for not equals.
> > + * @param comparison - The Id of a comparison to flip.
> > + * @return The flipped comparison id.
> > + */
> > +inline ComparisonID flipComparisonID(const ComparisonID comparison) {
>
> mark this function const.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pivotalsoftware/quickstep/pull/18/files#r51025581>.
>